vector/src/main/java/im/vector/app/features/home/room/detail/timeline/helper/MatrixItemColorProvider.kt

```diff
@@ -44,6 +46,42 @@ class MatrixItemColorProvider @Inject constructor(
         }
     }
 
+    fun setOverrideColors(overrideColors: Map<String, String>?) {
+        cache.clear()
+        overrideColors?.forEach {
+            setOverrideColor(it.key, it.value)
+        }
+    }
+
+    fun setOverrideColor(id: String, colorSpec: String?): Boolean {
+        val color = parseUserColorSpec(colorSpec)
+        return if (color == null) {
+            cache.remove(id)
+            false
+        } else {
+            cache[id] = color
+            true
+        }
+    }
+
+    @ColorInt
+    private fun parseUserColorSpec(colorText: String?): Int? {
+        return if (colorText.isNullOrBlank()) {
+            null
+        } else {
+            try {
+                if (colorText.length == 1) {
```

Comment:
This will break if/when the palette is extended to 16 or more colors.  Maybe it would be better to try just try to parse it as `int`, and if that works then it's a color index (from the palette) otherwise parse with `Color.parseColor`.  Another option would be to check if the text starts with a digit to select the palette version

Reply:
I agree, but the plan is to change this dialog to do something more user friendly, so it's acceptable like that to me.

Reply:
OK, cool, thanks.

Reply:
being able to set the colour by index is quite cryptic, I wouldn't have expected an input with a hex string hint to also allow a 0-9 index 

something a UI iteration could help to improve

Reply:
Yes, this is really a hidden feature. Sort of easter egg.
I see in my dream a nice dialog with all the 8 predefined colors, a color picker and a field to manually enter a html color. Even a reset button, instead of validating a empty dialog to reset the color.
Also note that now using the word "black" or any other color that `Color.parse()` understand will also work, not sure for the other non Android-Element Matrix clients compatibility though.